### DISCUSSION TITLE
Rename `__todo__`

### DISCUSSION THREAD

**Discussion by sampersand** [ruby/rbs#1607] [Ideas]
Added in RBS 3.2, `__todo__` is an alias for `untyped` which is intended to denote that a type hasn't been fleshed out yet.

I like the idea, but I'm not a fan of the name `__todo__`, as it breaks RBS's naming convention: It's a type alias (which should start with lowercase letters), but starts with a `_` like interfaces.

Ideas for solutions:
1. Rename to `todo__`. Not a huge fan of, because it looks a bit clunky
2. Rename to `todo`. Also not a huge fan, as this will probably conflict with user stuff
3. Change the RBS naming convention rules. Make it so type aliases are `^_*[a-z]\w*` and interfaces are `^_+[A-Z]\w*`.

I'm personally partial to the third option

**Comment by soutaro** [comment]
I don't think 1 or 2 is better than `__todo__`, because those names can be already defined in RBS files.

The third option would make sense, while `__todo__` cannot be a strong reason to do it (adding new keyword is more difficult if there is a chance to conflict with existing user defined names.)

**Reply by sampersand** [reply]
> I don't think 1 or 2 is better than __todo__, because those names can be already defined in RBS files.
Don't we already do this with things like `type int = _ToInt | Integer`?

<img width="652" alt="Screen Shot 2023-11-09 at 6 29 35 PM" src="https://github.com/ruby/rbs/assets/6655027/40306f46-621d-443c-87ab-3559a6957177">
This was the impetus: My syntax highlighter dies on `__todo__` (I should update it...), and I suspect others might too. Granted, we could make `__todo__` a keyword

**Reply by soutaro** [reply]
Released a new version of sublime package. https://github.com/soutaro/sublime-rbs-plugin/releases/tag/v1.1.0 So, it will fix soon.

**Comment by ParadoxV5** [comment]
`__TODO__`
